control-plane/pkg/reconciler/broker/controller.go

```diff
@@ -120,6 +123,19 @@ func NewController(ctx context.Context, watcher configmap.Watcher, configs *Conf
 		},
 	})
 
+	reconciler.SecretTracker = tracker.New(impl.EnqueueKey, controller.GetTrackerLease(ctx))
+	secretinformer.Get(ctx).Informer().AddEventHandler(controller.HandleAll(reconciler.SecretTracker.OnChanged))
+
+	reconciler.ConfigMapTracker = tracker.New(impl.EnqueueKey, controller.GetTrackerLease(ctx))
+	configmapinformer.Get(ctx).Informer().AddEventHandler(controller.HandleAll(reconciler.ConfigMapTracker.OnChanged))
+
+	brokerInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: kafka.BrokerClassFilter(),
+		Handler: cache.ResourceEventHandlerFuncs{
+			DeleteFunc: reconciler.OnDeleteObserver,
```

Comment:
Why that? Can't you just invoke `OnDeleteObserver` in the FinalizeKind of the broker reconciler?

Reply:
Same as the other comment.